src/bar.cpp

```diff
@@ -607,6 +590,26 @@ void waybar::Bar::handleSignal(int signal) {
   }
 }
 
+auto waybar::Bar::setVisible(bool nvis) -> void {
+  visible = nvis;
+  if (!visible) {
+    window.get_style_context()->add_class("hidden");
+    surface_impl_->setLayer(bar_layer::BOTTOM);
+  } else {
+    window.get_style_context()->remove_class("hidden");
+    surface_impl_->setLayer(bar_layer::TOP);
+  }
+  wl_surface_commit(surface);
```

Comment:
Extraneous `wl_surface_commit` is known to be able to crash Gtk in certain cases (as found and fixed in gtk-layer-shell). Also, surface can be unset you call that on an unmapped window.
That's why the code in master branch is calling `surface_impl_->commit()` which is noop for the most cases.